crates/pathfinder/src/storage/schema/revision_0010.rs

```diff
@@ -299,6 +309,73 @@ mod tests {
                     Ok(())
                 }
             }
+
+            #[derive(Clone, Debug, PartialEq)]
+            pub struct Transaction {
+                pub calldata: Option<Vec<CallParam>>,
+                pub class_hash: Option<ClassHash>,
+                pub constructor_calldata: Option<Vec<ConstructorParam>>,
+                pub contract_address: Option<ContractAddress>,
+                pub contract_address_salt: Option<ContractAddressSalt>,
+                pub entry_point_type: Option<EntryPointType>,
+                pub entry_point_selector: Option<EntryPoint>,
+                pub max_fee: Option<Fee>,
+                pub nonce: Option<TransactionNonce>,
+                pub sender_address: Option<ContractAddress>,
+                pub signature: Option<Vec<TransactionSignatureElem>>,
+                pub transaction_hash: StarknetTransactionHash,
+                pub r#type: Type,
+                pub version: Option<TransactionVersion>,
+            }
+
+            pub struct StarknetEventsTable {}
+            impl StarknetEventsTable {
+                pub fn insert_events(
+                    connection: &Connection,
+                    block_number: StarknetBlockNumber,
+                    transaction: &Transaction,
+                    events: &[transaction::Event],
+                ) -> anyhow::Result<()> {
+                    if transaction.contract_address.is_none() && !events.is_empty() {
+                        anyhow::bail!("Declare transactions cannot emit events");
+                    }
+
+                    for (idx, event) in events.iter().enumerate() {
+                        connection
+                        .execute(
+                            r"INSERT INTO starknet_events ( block_number,  idx,  transaction_hash,  from_address,  keys,  data)
+                                                   VALUES (:block_number, :idx, :transaction_hash, :from_address, :keys, :data)",
+                            named_params![
+                                ":block_number": block_number.0,
+                                ":idx": idx,
+                                ":transaction_hash": &transaction.transaction_hash.0.as_be_bytes()[..],
+                                ":from_address": &event.from_address.0.as_be_bytes()[..],
+                                ":keys": Self::event_keys_to_base64_strings(&event.keys),
+                                ":data": Self::event_data_to_bytes(&event.data),
+                            ],
+                        )
+                        .context("Insert events into events table")?;
+                    }
+                    Ok(())
+                }
+
+                pub fn event_data_to_bytes(data: &[EventData]) -> Vec<u8> {
```

Comment:
```suggestion
                fn event_data_to_bytes(data: &[EventData]) -> Vec<u8> {
```

Was curious to see where is this being used, it'd seem to be used from the tests. Oh but this is a migration, so probably not going to be used from anywhere...?